agora/Agora/Governor.hs

```diff
@@ -1,62 +1,791 @@
   GovernorRedeemer (..),
   Governor (..),
 
   -- * Plutarch-land
+  PGovernorDatum (..),
+  PGovernorRedeemer (..),
 
   -- * Scripts
   governorPolicy,
   governorValidator,
+
+  -- * Utilities
+  gstAssetClass,
+  gatSymbol,
 ) where
 
-import Agora.Proposal (ProposalId, ProposalThresholds)
+--------------------------------------------------------------------------------
+
+import GHC.Generics qualified as GHC
+import Generics.SOP (Generic, I (I))
+
+--------------------------------------------------------------------------------
+
+import Agora.AuthorityToken (
+  AuthorityToken (..),
+  authorityTokenPolicy,
+  authorityTokensValidIn,
+  singleAuthorityTokenBurned,
+ )
+import Agora.Proposal (
+  PProposalDatum (..),
+  PProposalId,
+  PProposalStatus (PFinished),
+  PProposalThresholds,
+  PResultTag,
+  Proposal (..),
+  ProposalId,
+  ProposalStatus (Draft, Locked),
+  ProposalThresholds,
+  pnextProposalId,
+  proposalDatumValid,
+  proposalPolicy,
+  proposalValidator,
+ )
+import Agora.SafeMoney (GTTag)
+import Agora.Stake (
+  PProposalLock (..),
+  PStakeDatum (..),
+  Stake (..),
+  stakePolicy,
+  stakeValidator,
+ )
+import Agora.Utils (
+  findOutputsToAddress,
+  hasOnlyOneTokenOfCurrencySymbol,
+  mustBePDJust,
+  mustBePJust,
+  mustFindDatum',
+  passert,
+  passetClassValueOf,
+  passetClassValueOf',
+  pfindDatum,
+  pfindTxInByTxOutRef,
+  pisDJust,
+  pisJust,
+  pisUxtoSpent,
+  pownCurrencySymbol,
+  psymbolValueOf,
+  ptxSignedBy,
+  pvalueSpent,
+  scriptHashFromAddress,
+ )
+
+--------------------------------------------------------------------------------
+
 import Plutarch (popaque)
-import Plutarch.Api.V1 (PMintingPolicy, PValidator)
+import Plutarch.Api.V1 (
+  PAddress,
+  PCurrencySymbol,
+  PDatumHash,
+  PMap,
+  PMintingPolicy,
+  PScriptPurpose (PSpending),
+  PTxOut,
+  PValidator,
+  PValidatorHash,
+  PValue,
+  mintingPolicySymbol,
+  mkMintingPolicy,
+  mkValidator,
+  validatorHash,
+ )
+import Plutarch.Api.V1.Extra (
+  pownMintValue,
+ )
+import Plutarch.Builtin (pforgetData)
+import Plutarch.DataRepr (
+  DerivePConstantViaData (..),
+  PDataFields,
+  PIsDataReprInstances (PIsDataReprInstances),
+ )
+import Plutarch.Lift (PUnsafeLiftDecl (..))
+import Plutarch.Map.Extra (pkeys, plookup, plookup')
+import Plutarch.Monadic qualified as P
+import Plutarch.SafeMoney (PDiscrete, Tagged (..), puntag, pvalueDiscrete)
+import Plutarch.Unsafe (punsafeCoerce)
+
+--------------------------------------------------------------------------------
+
+import Plutus.V1.Ledger.Api (
+  Address (Address),
+  Credential (ScriptCredential),
+  CurrencySymbol (..),
+  MintingPolicy,
+  TxOutRef,
+ )
+import Plutus.V1.Ledger.Value (
+  AssetClass (..),
+ )
+import PlutusTx qualified
+
+--------------------------------------------------------------------------------
+
+{- $gst
+   Governance state token, aka. GST, is an NFT that identifies an UTXO that carries the state datum of the Governance script.
```

Comment:
'a UTXO'. If you say 'UTXO' slowly, you see it starts with a 'y' sound. As 'y' is a consonant, it should be `a UTXO`. As a counter-point, you wrote 'an NFT' which is correct, as 'NFT' starts with an 'e' sound :D